bin/src/main.rs

```diff
@@ -118,26 +118,65 @@ async fn main() {
     }
 
     let mut executor = Executor::new();
-    if config.log.use_k8s_enrichment == K8sTrackingConf::Always
-        && PathBuf::from("/var/log/containers/").exists()
-    {
-        let node_name = std::env::var("NODE_NAME").ok();
-        match K8sMetadata::new(user_agent, node_name.as_deref()).await {
-            Ok((driver, v)) => {
-                tokio::spawn(driver);
-                executor.register(v);
-                info!("Registered k8s metadata middleware");
-            }
-            Err(e) => {
-                let message = format!(
-                    "The agent could not access k8s api after several attempts: {}",
-                    e
-                );
-                error!("{}", message);
-                panic!("{}", message);
+
+    let k8s_event_stream = match create_k8s_client_default_from_env(user_agent) {
+        Ok(k8s_client) => {
+            if config.log.use_k8s_enrichment == K8sTrackingConf::Always
+                && PathBuf::from("/var/log/containers/").exists()
```

Comment:
using hardcoded /var/log/containers, and few other places
side note: 
files in /var/log/containers are symlinks to files in /var/log/pods. /var/log/pods is primary location. would it be better to switch to /var/log/pods? yes, it will require patching the log file name parsing in k8s enrichment, but overall will more stable solution going forward.

Reply:
I don't have a strong opinion, We've always using /var/log/containers as the agent doesn't really distinguish between symlinks and files, they're all reduced down to the actual inode containing the data